docs/msi-mpg-coreliquid-guide.md

```diff
@@ -0,0 +1,165 @@
+| `color pulse` | ? | ? | Not working as intended |
+| `color ring` | None | None |                        |
+| `color ring double flashing` | None | None |                        |
+| `color ring flashing` | None | None |                        |
+| `color shift` | None | None | Not working as intended  |
+| `color wave` | Two | Yes |                        |
+| `corsair ique` | ? | ? | Unclear   |
+| `disable2` | None | None |                        |
+| `double flashing` | One | Yes |                        |
+| `double meteor` | None | None |                        |
+| `energy` | None | None |                        |
+| `fan control` | ? | ? | Not working as intended |
+| `fire` | Two | No |                        |
+| `flashing` | One | Yes |                        |
+| `jazz` | ? | ? | Not working as intended |
+| `jrainbow` | ? | ? | Unclear |
+| `lava` | ? | ? | Not working as intended |
+| `lightning` | One | No |                        |
+| `marquee` | One | No |                        |
+| `meteor` | One | Yes |                        |
+| `movie` | ? | ? | Not working as intended |
+| `msi marquee` | One | Yes |                        |
+| `msi rainbow` | ? | ? | Not working as intended |
+| `planetary` | None | None |                        |
+| `play` | ? | ? | Not working as intended |
+| `pop` | ? | ? | Not working as intended |
+| `rainbow` | ? | ? | Very jittery and slow, rainbow wave is recommended instead |
+| `rainbow double flashing` | None | None |                        |
+| `rainbow flashing` | None | None |                        |
+| `rainbow wave` | None | None |                        |
+| `random` | None | None |                        |
+| `rap` | ? | ? | Not working as intended |
+| `stack` | One | Yes |                        |
+| `visor` | Two | Yes |                        |
+| `water drop` | One | Yes |                        |
+
+Support for the rainbow option is not yet exposed by the liquidctl cli, but it is included in the driver.
+
+## The LCD screen
+
+The screen resolution is 320 x 240 px, and custom images uploaded with this driver are resized to fit this requirement. The screen orientation and brightness (0-100) can also be controlled. The only channel available for the K360 model is "lcd".
+
+Maximum length of the displayed banner mesages is 62 ASCII characters. hardware status display functionality is limited, as the displayed data must be communicated to the device. This functionality is implemented in the driver, but currently its usage is limited to yoda, which gpu-unaware so the gpu_freq and gpu_usage parameters will not display correct information without custom update services.
+
+
+| mode name | action | options |
+| --- | --- | --- |
+| hardware | set the screen to display hardware info | up to 3 semicolon delimited keys from the available sensors |
+| image | set the screen to display a custom or preset image | \<type (0=preset,1=custom)\>;\<index\>[;\<filename\>] |
```

Comment:
The `<index>` refers to a save slot on the device. It has several preset animations and is supposed to be able to store multiple custom images that can be selected with their uploaded index. This has been the most challenging part to get working, custom images now still seem to overwrite any previous custom images no matter which index is used, but the index at which the custom image can be loaded is also specific, and inconsistent. Often, but not always, the first time after uploading an image, loading its index will instead load a preset, and sometimes, but not always, the custom image will also get viewed from other indices preserved for the custom files in addition to the one it was uploaded to.
Trying to upload an image with the type flag 0 currently raises an error. I thought this would be the clearest option syntax, the layout lends itself more naturally to MSI's GUI where you only have separate rows for preset and custom images. The redundant `1` could be prevented with another flag specifically for `upload-image`, but there are already many enough to be slightly confusing. I am open to alternative cleaner solutions though.

Reply:
> Or is it just for `<direction (0-3)>` in case of `settings`?
Yes.

Reply:
I added some clarification to the docs about this now.